operator/controllers/seldondeployment_controller.go

```diff
@@ -310,13 +328,48 @@ func createIstioResources(mlDep *machinelearningv1.SeldonDeployment,
 					Number: uint32(ports[i].grpcPort),
 				},
 			},
-			Weight: p.Traffic,
+		}
+
+		if p.Traffic != 0 {
+			routesHttp[routesIdx].Weight = p.Traffic
+			routesGrpc[routesIdx].Weight = p.Traffic
+		}
+
+		// set traffic by http header
+		for _, match := range mlDep.Spec.Predictors[i].TrafficMatchs {
+			isContentTraffic = true
+
+			headers := make(map[string]*istio_networking.StringMatch)
+
+			for key, value := range match.Headers {
+				stringMatchPtr := machinelearningv1.ConvertIstioStringMatch(value)
+				if stringMatchPtr == nil {
+					continue
+				}
+				headers[key] = stringMatchPtr
+			}
+
+			// For http
+			httpRoute.Match[0].Headers = mergeIstioMatchHeaders(httpRoute.Match[0].Headers, headers)
```

Comment:
Should we check the length of match here?

Reply:
SGTM, it is addressed